lib/rubocop/cop/rails/date.rb

```diff
@@ -0,0 +1,124 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for the correct use of Date methods,
+      # such as Date.today, Date.current etc.
+      #
+      # Using Date.today is dangerous, because it doesn't know anything about
+      # Rails time zone. You must use Time.zone.today instead.
+      #
+      # The cop also reports warnings when you are using 'to_time' method,
+      # because it doesn't know about Rails time zone too.
+      #
+      # Two styles are supported for this cop. When EnforcedStyle is 'always'
+      # then the Date methods (today, current, yesterday, tomorrow)
+      # are prohibited and the usage of both 'to_time'
+      # and 'to_time_in_current_zone' is reported as warning.
+      #
+      # When EnforcedStyle is 'acceptable' then only 'Date.today' is prohibited
+      # and only 'to_time' is reported as warning.
+      #
+      # @example
+      #   # no offense
+      #   Time.zone.today
+      #   Time.zone.today - 1.day
+      #
+      #   # acceptable
+      #   Date.current
+      #   Date.yesterday
+      #
+      #   # always reports error offense
+      #   Date.today
```

Comment:
The error type is reserved for syntax errors, so this should report a warning as well.